src/api/exchange/ExchangeApi.ts

```diff
@@ -296,6 +426,76 @@ export class ExchangeApiImpl extends DepositApiImpl implements ExchangeApi {
 
     return tx
   }
+
+  /** PRIVATE methods **/
+
+  private async safeGetEvents<T>(
+    fn: (options: PastEventOptions) => Promise<T[]>,
+    options: PastEventOptions,
+  ): Promise<T[]> {
+    try {
+      return await fn(options)
+    } catch (e) {
+      // Error `-32005` means too many results in range.
+      // Let's split it up into 2 smaller requests
+      if (e.code === -32005) {
```

Comment:
Cool, should we move it to a const and maybe add a link to this https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1474.md